Userland/Libraries/LibGfx/MedianCut.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2024, Lucas Chollet <lucas.chollet@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/HashMap.h>
+#include <AK/Vector.h>
+#include <LibGfx/Bitmap.h>
+#include <LibGfx/Color.h>
+
+namespace Gfx {
+
+class ColorPalette {
+public:
+    struct ColorAndIndex {
+        Color color;
+        size_t index;
+    };
+
+    ColorPalette(Vector<Color> palette, HashMap<Color, ColorAndIndex> conversion_table)
+        : m_palette(move(palette))
+        , m_conversion_table(move(conversion_table))
+    {
+    }
+
+    Vector<Color> const& palette() const
+    {
+        return m_palette;
+    }
+
+    Color closest_color(Color input) const
+    {
+        return m_palette[index_of_closest_color(input)];
+    }
+
+    u32 index_of_closest_color(Color input) const
+    {
+        if (auto const result = m_conversion_table.get(input); result.has_value())
+            return result->index;
+        TODO();
```

Comment:
😬

Doing some kind of error diffusion dithering would probably be nice, which this type of API isn't amenable to. But, baby steps etc

Reply:
From what I understand, dithering is always done as a second step after finding the palette, so this is a separated issue. The code that needs to be added here is a code that find the closest color in the palette from any input color. Anyway, both of these things are for later.